### PR TITLE
Avoid setting global numpy PRNG seed

### PR DESCRIPTION
**Pull request by mcwitt**
Since #1294, the default HREX implementation no longer relies on numpy global PRNG state. This PR modifies the signature of `attempt_neighbor_swaps_fast` to accept an explicit seed rather than generating one using `numpy.random` and removes an associated warning.

Minor unrelated change: explicitly declare integer dtypes for proposed and accepted count arrays

### CODE REVIEW

**Comment by proteneer** [comment]
awesome!

File: timemachine/fe/free_energy.py [hunk 1]
```diff
@@ -1099,7 +1094,7 @@ def replica_from_samples(traj: Trajectory) -> CoordsVelBox:
         hrex, samples_by_state_iter = hrex.sample_replicas(sample_replica, replica_from_samples)
         log_q_kl = compute_log_q_matrix(hrex.replicas)
         hrex, fraction_accepted_by_pair = hrex.attempt_neighbor_swaps_fast(
-            neighbor_pairs, log_q_kl, n_swap_attempts_per_iter
+            neighbor_pairs, log_q_kl, n_swap_attempts_per_iter, md_params.seed
```
Comment (badisa) [inline timemachine/fe/free_energy.py:1097]: Should this do `md_params.seed + iteration` to avoid generating the same sequence every time?

Comment (mcwitt) [reply timemachine/fe/free_energy.py:1097]: Yes, I think so! Good catch.

**Comment by mcwitt** [comment]
Also noticed an unrelated bug (fixed in https://github.com/proteneer/timemachine/pull/1302/commits/4587c3d7932ae1075c429e15c6b088ba0ba1ed07) that I think resulted in `water_sampling_params` always being `None` in the test

File: timemachine/md/hrex.py [hunk 2]
```diff
@@ -426,6 +423,9 @@ def run_hrex(
     n_samples_per_iter: int
         Number of local samples (e.g. MD frames) per HREX iteration
 
+    seed : int
```
Comment (badisa) [inline timemachine/md/hrex.py:426]: super nit:
```suggestion
    seed: int
```

File: timemachine/md/hrex.py [hunk 3]
```diff
@@ -213,13 +206,16 @@ def attempt_neighbor_swaps_fast(
         n_swap_attempts : int
             number of individual swap attempts, each between a randomly-selected neighbor pair
 
+        seed : int
```
Comment (badisa) [inline timemachine/md/hrex.py:209]: super nit:
```suggestion
        seed: int
```

**Review by badisa — APPROVED** [review]
Nice, glad you cause that bug in the test.

File: timemachine/md/hrex.py (hunk 3 above)
Comment (mcwitt) [reply timemachine/md/hrex.py:209]: It seems we have a mixed convention. Fwiw, the numpydoc style guide includes the extra space: https://numpydoc.readthedocs.io/en/latest/format.html

Maybe I'll update so we're at least consistent within a given docstring for now?

**Review by maxentile — APPROVED** [review]

File: timemachine/md/hrex.py (hunk 3 above)
Comment (mcwitt) [reply timemachine/md/hrex.py:209]: I applied the other suggested fix. Will leave this one as-is for now for consistency.

**Pull request merged**